OpenRVS/classes/OpenHTTPClient.uc

```diff
@@ -0,0 +1,292 @@
+// OpenHTTPClient handles all HTTP/TCP traffic originating from the game client.
+// The base game only uses UDP, so this is only used by us.
+// TcpLink extends InternetLink extends InternetInfo extends Info extends Actor.
+// Use Spawn(class'OpenHTTPClient') to get an instance.
+// WARNING: UE2 only supports HTTP 1.0; HTTP 1.1 connections won't close before
+// timing out.
+// Instructions for using OpenHTTPClient:
+// 1. Write a callback function with the signature `function C(HttpResponse r)`.
+//    Your function can operate on the status code `r.Code` or the response
+//    body `r.Body`. This function should be added to the very bottom of this
+//    file.
+// 2. Add the name of your callback as a const in this class.
+// 3. Add a reference to your callback provider as a var in this class.
+// 4. Register your callback in the switch statement under Closed() in this class.
+// 5. After spawning the client, set c.CallbackName = "your_callback" and set
+//    c.YourCallbackProvider = class. Then you can call c.SendRequest() and
+//    receive the data in your callback provider.
+class OpenHTTPClient extends TcpLink;
+
+//
+// Types
+//
+
+struct ParsedURL
+{
+	var string Host;//e.g. google.com
+	var int Port;//e.g. 80
+	var string Path;//e.g. /servers
+};
+
+struct HttpResponse
+{
+	var int Code;//e.g. 200 (OK) or 404 (Not Found)
+	var string Body;//response body
+};
+
+//
+// Instance Variables
+//
+
+var HttpResponse Response;
+var array<string> kiloBuffer;//array of 999-byte response segments
+var bool requestInFlight;
+
+var string Host;
+var int Port;
+var string Path;
+
+const CALLBACK_SERVER_LIST = "server_list";
+var string CallbackName;
+var OpenServerList ServerListCallbackProvider;
+// Add new Callback vars here.
+
+//
+// Event Overrides
+//
+
+// Called when domain resolution is successful.
+// The IpAddr struct Addr contains the valid address.
+event Resolved(IpAddr Addr)
+{
+	super.Resolved(Addr);
+
+	class'OpenLogger'.static.Debug("dns resolution succeeded, initiating connection", self);
+	Addr.Port = Port;
+	Open(Addr);	
+}
+
+// Called when domain resolution fails.
+event ResolveFailed()
+{
+	super.ResolveFailed();
+	super.Destroy();//clean up the connection
+
+	class'OpenLogger'.static.Error("dns resolution failed", self);
+}
+
+// Opened: Called when socket successfully connects.
+// Sends an HTTP request.
+event Opened()
+{
+	local string CRLF;
+
+	super.Opened();
+	class'OpenLogger'.static.Debug("connection opened", self);
+
+	// HTTP GET request format:
+	// First line is "GET $PATH $HOST HTTP/1.1" followed by CRLF.
```

Comment:
We should have this comment match the 1.0 line instead of keeping the 1.1 ;)

Reply:
Updated